ceph/rados/core_workflows.py

```diff
@@ -4615,3 +4616,56 @@ def get_rados_df(self, pool_name: str = None):
         out = self.run_ceph_command(cmd=_cmd, client_exec=True)
 
         return out["pools"][0] if pool_name else out
+
+    def set_service_managed_type(self, service_type, unmanaged) -> bool:
+        """
+        Method to set the service to either managed or unmanaged
+        Args:
+            unmanaged: True or false, for the service management
+            service_type : service types are- mon,mgr,osd,rgw, mds
+        returns:
+            Pass -> True, Fail -> false
+        """
+        cmd_export = f"ceph orch ls {service_type} --export"
+        out = self.run_ceph_command(cmd=cmd_export, client_exec=True)[0]
+        if unmanaged:
+            log.debug(
+                f"Setting the {service_type} service as unmanaged by cephadm. current status : {out}"
+            )
+            out["unmanaged"] = "true"
+        else:
+            log.debug(
+                f"Setting the {service_type} service as unmanaged by cephadm. current status : {out}"
+            )
+            out["unmanaged"] = "false"
+
+        file_name = (
+            f"/tmp/{service_type}_spec_{self.set_service_managed_type.__name__}.yaml"
+        )
+        # Creating service config file
+        self.client.exec_command(sudo=True, cmd=f"touch {file_name}")
+        json_out = json.dumps(out)
+        # Adding the spec rules into the file
+        cmd = f"echo '{json_out}' > {file_name}"
+        self.client.exec_command(cmd=cmd, sudo=True)
+
+        log.debug(f"Contents of {service_type} spec file : {out}")
+        apply_cmd = f"ceph orch apply -i {file_name}"
+        log.info(f"Applying the spec file via cmd : {apply_cmd}")
+        self.client.exec_command(cmd=apply_cmd, sudo=True)
+        time.sleep(10)
+        out = self.list_orch_services(service_type="osd", export=True)[0]
```

Comment:
Same here. iterate through all the elements for the services.

Reply:
The same as the above-
 /home/skanta/cephci_tfa/cephci/ceph/rados/core_workflows.py(4661)set_service_managed_type()
-> out = self.list_orch_services(service_type=service_type, export=True)[0]
(Pdb) n
2024-12-02 05:31:39,811 (cephci.test_delete) [INFO] - cephci.ceph.ceph.py:1570 - Execute ceph orch ls osd --export -f json on 10.0.64.178
2024-12-02 05:31:40,825 (cephci.test_delete) [INFO] - cephci.ceph.ceph.py:1600 - Execution of ceph orch ls osd --export -f json on 10.0.64.178 took 1.005422 seconds.
> /home/skanta/cephci_tfa/cephci/ceph/rados/core_workflows.py(4662)set_service_managed_type()
-> out_test = self.list_orch_services(service_type=service_type, export=True)
(Pdb) n
2024-12-02 05:31:43,029 (cephci.test_delete) [INFO] - cephci.ceph.ceph.py:1570 - Execute ceph orch ls osd --export -f json on 10.0.64.178
2024-12-02 05:31:44,039 (cephci.test_delete) [INFO] - cephci.ceph.ceph.py:1600 - Execution of ceph orch ls osd --export -f json on 10.0.64.178 took 1.005566 seconds.
> /home/skanta/cephci_tfa/cephci/ceph/rados/core_workflows.py(4663)set_service_managed_type()
-> status = out.get("unmanaged", False)
(Pdb) p out
{'placement': {'host_pattern': '*'}, 'service_id': 'all-available-devices', 'service_name': 'osd.all-available-devices', 'service_type': 'osd', 'spec': {'data_devices': {'all': True}, 'filter_logic': 'AND', 'objectstore': 'bluestore'}, 'unmanaged': 'false'}
(Pdb) p out_test
[{'placement': {'host_pattern': '*'}, 'service_id': 'all-available-devices', 'service_name': 'osd.all-available-devices', 'service_type': 'osd', 'spec': {'data_devices': {'all': True}, 'filter_logic': 'AND', 'objectstore': 'bluestore'}, 'unmanaged': 'false'}]
(Pdb) p type(out)
<class 'dict'>
(Pdb) p type(out_test)
<class 'list'>
(Pdb)

Reply:
Modified the code as pre the comments.